libr/core/canal.c

```diff
@@ -4036,9 +4042,15 @@ R_API void r_core_anal_esil(RCore *core, const char *str, const char *target) {
 		}
 		switch (op.type) {
 		case R_ANAL_OP_TYPE_SWI:
-			if (!refptr && (in == -1 || op.val == in)) {
+
+			if (!refptr && (in == -1 || op.val == in || canal_isThumb (core))) {
```

Comment:
Dont love this, since we check again below for canal_isThumb as well.. But otherwise the whole thing probably needs a refactor...

Reply:
Me neither. In fact the check for isthumb should be deleted completely